plugin/selection_range.py

```diff
@@ -0,0 +1,55 @@
+from .core.protocol import Range
+from .core.protocol import Request
+from .core.registry import get_position
+from .core.registry import LspTextCommand
+from .core.sessions import method_to_capability
+from .core.typing import Any, Dict, Optional, List
+from .core.views import range_to_region
+from .core.views import selection_range_params
+import sublime
+
+
+class LspExpandSelectionCommand(LspTextCommand):
+    method = 'textDocument/selectionRange'
+    capability = method_to_capability(method)[0]
+
+    def __init__(self, view: sublime.View) -> None:
+        super().__init__(view)
+        self._regions = []  # type: List[sublime.Region]
+        self._change_count = 0
+
+    def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
+        session = self.session(self.capability, get_position(self.view, event))
+        if session:
+            params = selection_range_params(self.view)
+            self._regions.extend(self.view.sel())
+            self._change_count = self.view.change_count()
+            session.send_request(Request(self.method, params), self.on_result, self.on_error)
+
```

Comment:
We need to trigger original key binding if there is no session for position. Otherwise expanding will just be broken when there is a session but it doesn't support expanding.

Reply:
Hmm, I tried to address this but I learned that the `is_enabled` method from LspTextCommand prevents the keybinding to trigger. But it also masks the original `expand_selection` command in this way (you cannot trigger it with ctrl+shift+a anymore, because another command (lsp_expand_selection) masked it with its is_enabled method returning False). So I've added the fallback behavior, but it doesn't do anything at this point. Implementing #1093 will make things work better anyway.

Reply:
I have removed the ctrl+shift+a suggestion from the default keybindings for now.

Reply:
Although I see that this would not be ideal for the context menu.

Reply:
Yes, it will not be grayed out in the context menu when we always return True from is_enabled.

Reply:
It might be OK in fact. Sure, it wouldn't necessarily use the LSP to expand the selection, but if user has made the effort to look for it in the submenu, maybe he should just be able to expand, regardless if it's done by LSP or ST.